### PR TITLE
Feature/group page final

### PR DESCRIPTION
**Pull request by trevorfitz0**
## Neccesary checkmarks:

    [x] All Tests are Passing in all environments

    [x] The code will run locally

## Type of change

    [x] New feature
    [] Bug Fix
    [x] Refactor

## Description of Change:

    description: Set up group page for post requests and changed code to allow user access after the change
    
## Requested feedback: 
    
    I'd like feedback on... 

## Check the correct boxes

    [x] This broke nothing
    [] This broke some stuff
    [] This broke everything

## Testing Changes

    [] No Tests have been changed
    [x] Some Tests have been changed
    [] All of the Tests have been changed(Please describe what in the world happened)

## Checklist:

    [x] My code has no unused/commented out code
    [] My code has no binding.pry's
    [x] I have reviewed my code
    [] I have commented my code, particularly in hard-to-understand areas
    [x] I have fully tested my code

### CODE REVIEW

File: cypress/fixtures/groups.json [hunk 1]
```diff
@@ -1,17 +1,36 @@
...
+        "id": 3,
+        "name": "Super cool fun times",
+        "updated": "2023-05-24T19:18:01.675727Z",
+        "created": "2023-05-24T19:18:01.675749Z",
+        "user": 4,
+        "friends": [
+          3
+        ]
+      }
     ]
```
Comment (Gderibeaux) [inline cypress/fixtures/groups.json:35]: Nice I like extra details

File: cypress/support/commands.js [hunk 2]
```diff
@@ -45,9 +46,23 @@ Cypress.Commands.add('clickEventOne', () => {
...
+    cy.intercept('POST', 'https://lynk-up-server.onrender.com/groups/create', (req) => {
+      req.reply({
+        status: 200,
+        body: {
+          "name": "awesome fun group!",
+          "user": "1",
+          "friends": "[1]"
+      }
+      })
+    })
```
Comment (Gderibeaux) [inline cypress/support/commands.js:67]: Nice work!

**Review by Gderibeaux — APPROVED** [review]
Everything looks good we will probably have some merge conflicts once I push mine but everything looks correct

File: cypress/e2e/group-spec.cy.js [hunk 3]
```diff
@@ -22,7 +22,19 @@ describe('Group Page', () => {
...
     it('Should be able to see a list of groups', () => {
-        cy.get('.groupPage > :nth-child(1) > :nth-child(5)').should('contain', 'Group 273?');
+        cy.get('.groupPage > :nth-child(1) > :nth-child(5)').should('contain', 'Super cool fun times');
+    });
+
+    it('Should be able to post a new group', () => {
+        cy.get('[placeholder="Group Name"]').type("awesome fun group!");
+        cy.get('.form > :nth-child(4)').click();
+
+        cy.postGroup();
```
Comment (jfogiato) [inline cypress/e2e/group-spec.cy.js:32]: Great use of cypress Commands!

File: src/ApiCalls.js [hunk 4]
```diff
@@ -45,7 +44,29 @@ const getFriends = (user_id) => {
...
+    fetch("https://lynk-up-server.onrender.com/groups/create/", {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({
+        "name": groupInfo.name,
+        "user": groupInfo.user,
+        "friends": groupInfo.friends
+      }),
```
Comment (jfogiato) [inline src/ApiCalls.js:60]: Super solid work, thanks Trevor!

**Pull request merged**